service/service.go

```diff
@@ -71,8 +74,14 @@ func (s *Service) ValidateEventData(eventKey string, eventData map[string]interf
 }
 
 // RequireTaskInputs requires task inputs to match with parameter schemas.
-func (s *Service) RequireTaskInputs(taskKey string, taskInputs map[string]interface{}) error {
-	warnings, err := s.ValidateTaskInputs(taskKey, taskInputs)
+func (s *Service) RequireTaskInputs(taskKey string, taskInputs *types.Struct) error {
+	in := make(map[string]interface{})
+	if taskInputs != nil {
+		if err := convert.Marshal(taskInputs, &in); err != nil {
+			return err
+		}
+	}
+	warnings, err := s.ValidateTaskInputs(taskKey, in)
```

Comment:
will it be possible to update the function `ValidateTaskInputs` and co to accept directly `*types.Struct`?

Reply:
I guess it will be possible and maybe even better for the type casting as `struct.Value` seems to already manage type.